doc/doxygen/src/mainpage.md

```diff
@@ -0,0 +1,244 @@
+
+To use RIOT directly on your embedded platform, and for more hands-on details
+with RIOT, see [TODO: link to getting started guide].
+
+Before that, skimming through the next section is recommended (but not
+mandatory).
+
+Structure                                                          {#structure}
+=========
+
+This section walks you through RIOT's structure. Once you understand this
+structure, you will easiy find your way around in RIOT's code base.
+
+RIOT's code base is structured into five groups.
+
+<!-- TODO: add graphic -->
+
+ - The kernel (`core`)
+ - Platform specific code (`cpu`; `boards`)
+ - Device drivers (`drivers`)
+ - Device drivers (`drivers`)
+ - Libraries and network code (`sys/pkg`)
+ - Applications for demonstrating features and for testing (`examples`;
+   `tests`)
+
+In addition RIOT includes a collection of scripts for various tasks (`dist`) as
+well as a predefined environment for generating this documentation (`doc`).
+
+These groups are projected onto the folder structure of RIOT, where each of
+these groups resides in one or two folders in the main RIOT folder.
+
+The following list gives a more detailed description of each of RIOT's
+top-level folders:
+
+core
+----
+This folder contains the actual kernel. The kernel consists of the scheduler,
+inter-process-communication (messaging), threading, thread synchronization, and
+supporting data-structures and type definitions.
+
+See @ref core for further information and API documentations.
+
+boards
+------
+The platform dependent code is split into two logic elements: CPU's and boards,
+while maintaining a strict 1-to-n relationship, a board has exactly one CPU,
+while a CPU can be part of n boards. The CPU part contains all generic, CPU
+specific code (see below).
+
+The board part contains the specific configuration for the CPU it contains.
+This configuration mainly includes the peripheral configuration and
+pin-mapping, the configuration of on-board devices, and the CPU's clock
+configuration.
+
+On top of the source and header files needed for each board, this folder
+additionally may include some script and configuration files needed for
+interfacing with the board. These are typically custom flash/debug scripts or
+e.g. OpenOCD configuration files. For most boards, these files are located in a
+`dist` sub-directory of the board.
+
+See here @ref boards for further information.
+
+cpu
+---
+For each supported CPU this folder contains a sub-directory with the name of
+the CPU. These folders then contain all CPU specific configurations, such as
+implemenations of power management (LPM), interrupt handling and vectors,
+startup code, clock inialization code and thread handling (e.g. context
+switching) code.  For most CPU's you will also find the linkerscripts in the
+`ldscripts`
+sub-directory.
+
+In the `periph` sub-directory of each CPU you can find the implementations of
+the CPU's peripheral drivers like SPI, UART, GPIO, etc.  See @ref driver_periph
```

Comment:
> for each CPU there is the implementation of the peripheral drivers of the CPU 

vs

> for each CPU there is the implementation of the peripheral drivers of the CPUs

Logically the first statement makes more sense, so singular genitive makes more sense